### DISCUSSION TITLE
Having trouble using assembly attribute Xunit.TestFramework with type from another assembly

### DISCUSSION THREAD

**Discussion by thecristidima** [xunit/xunit#2775] [Question] [answered]
I have two sample projects - `TestProject` and `TestLibrary`. In the library I have this custom implementation of `XunitTestFramework`

```
namespace TestLibrary
{
    public class CustomTestFramework : XunitTestFramework
    {
        public CustomTestFramework(IMessageSink diagnosticMessageSink) : base(diagnosticMessageSink)
        {
            File.WriteAllText(@"C:\bla\library.txt", DateTime.UtcNow.ToShortTimeString()); // just to make sure this does anything
        }
    }
}
```

In my test project in my single test file I have this:

`[assembly: Xunit.TestFramework("TestLibrary.CustomTestFramework", "TestProject")]`

This setup doesn't work and I was wondering why. I should mention that moving `CustomTestFramework` to the test project and changing the type name works, but what I'm trying to do is use this class in several test projects without adding any new files in those projects.

**Comment by bartelink** [comment, marked as answer]
>`"TestLibrary.CustomTestFramework", "TestProject`

Is it possible that `"TestProject"` here should be `"TestLibrary"` ? (its obv possible this is just a mispaste as you were trying different things)

**Reply by thecristidima** [reply]
Yep, that was it. For some reason I thought the second parameter was the current assembly, which obviously makes no sense :sweat_smile:
